### PR TITLE
guard make check, ctest and update cmake args

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

File: conda_forge_tick/migrators/cross_compile.py [hunk 1]
```diff
@@ -81,3 +83,54 @@ def migrate(self, recipe_dir: str, attrs: "AttrsTypedDict", **kwargs: Any) -> No
...
+class GuardTestingMigrator(CrossCompilationMigratorBase):
+    def migrate(self, recipe_dir: str, attrs: "AttrsTypedDict", **kwargs: Any) -> None:
+        with indir(recipe_dir):
+            if not os.path.exists("build.sh"):
+                return
+            with open("build.sh", "r") as f:
+                lines = list(f.readlines())
+
+            for i, line in enumerate(lines):
+                if line.startswith("make check") or line.startswith("ctest"):
```
Comment (beckermr) [inline conda_forge_tick/migrators/cross_compile.py:97]: do we want to look for `make test` too?

**Comment by CJ-Wright** [comment]
I think you also need to update the osx migrator to support minimigrators

File: conda_forge_tick/auto_tick.py [hunk 2]
```diff
@@ -401,7 +403,12 @@ def add_arch_migrate(migrators: MutableSequence[Migrator], gx: nx.DiGraph) -> No
             graph=total_graph,
             pr_limit=PR_LIMIT,
             name="arm osx addition",
-            piggy_back_migrations=[UpdateConfigSubGuessMigrator()],
+            piggy_back_migrations=[
```
Comment (isuruf) [inline conda_forge_tick/auto_tick.py:406]: @CJ-Wright, is this not enough?

Comment (CJ-Wright) [reply conda_forge_tick/auto_tick.py:406]: The class currently doesn't take that as an arg.

**Comment by isuruf** [comment]
Ready for another review

**Pull request merged**